internal/handlers/images.go

```diff
@@ -67,39 +74,10 @@ func (h *ImageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	params := r.URL.Query()
-
-	version := params.Get("version")
-	if version == "" {
-		w.WriteHeader(http.StatusBadRequest)
-		_, err = w.Write([]byte("'version' parameter required"))
-		if err != nil {
-			log.Errorf("Failed to write response: %v\n", err)
-		}
-		return
-	}
-
-	if !h.ImageStore.HaveVersion(version) {
-		w.WriteHeader(http.StatusBadRequest)
-		message := fmt.Sprintf("version %s not found", version)
-		_, err = w.Write([]byte(message))
-		if err != nil {
-			log.Errorf("Failed to write response: %v\n", err)
-		}
-		return
-	}
-
-	imageType := params.Get("type")
-	if imageType == "" {
-		w.WriteHeader(http.StatusBadRequest)
-		_, err = w.Write([]byte("'type' parameter required"))
-		if err != nil {
-			log.Errorf("Failed to write response: %v\n", err)
-		}
-		return
-	} else if imageType != imagestore.ImageTypeFull && imageType != imagestore.ImageTypeMinimal {
+	version, imageType, apiKey, err := h.getQueryParams(r.URL.Query())
```

Comment:
`getQueryParams` name mislead me a bit, maybe `parseQueryParams`? `getQueryParams` sounds a bit like you're constructing query params for a request you want to make to a different service